docs/okeps/okep-5088-evpn.md

```diff
@@ -0,0 +1,789 @@
+across the external network fabric. 
+Finally, EVPN is a common datacenter networking fabric that many users with Kubernetes clusters already
+rely on for their top of rack (TOR) network connectivity. It is a natural next step to enable the Kubernetes platform
+to be able to directly integrate with this fabric directly.
+
+### EVPN/UDN Background
+
+This section provides some background on UDN, EVPN. It is important to provide this context for a better understanding of
+the following sections and design choices.
+
+UDNs are separate and potentially overlapping networks that are naturally isolated from each other. Each Primary UDN maps
+to a corresponding VRF in Linux. This is true for both Layer 2 and Layer 3 networks. The ovn-k8s-mpx interface exists
+per UDN and is configured inside of this Linux VRF.
+
+EVPN provides a mechanism for carrying potentially overlapping Layer 2 and/or Layer 3 networks while keeping them
+isolated from each other in a Virtual Private Network (VPN) over an ethernet fabric via BGP extensions and an overlay
+mechanism, typically VXLAN.
+
+Therefore, it is a logical next step to leverage EVPN to carry PUDNs as it fits the paradigm well. However, there are key
+design choices to be aware of. With EVPN a Layer 2 network (MAC-VRF) can be thought of as simply a switch extended across
+the ethernet fabric and used for East/West traffic. 
+With EVPN a Layer 2 network can belong to MAC-VRF as well as an IP-VRF (Layer 3) network simultaneously.
+In practice, this means the Layer 2 switch has a Switch Virtual Interface (SVI) which is plugged into a router (VRF), and
+clients on the Layer 2 network use this SVI as their default gateway for routing. In other words, if you have a pod connected
+to a switch (MAC-VRF that spans all nodes) it can talk to other pods or entities on that Layer 2 network without leaving the
+MAC-VRF. On the other hand if this pod needs to talk to the internet, it will go via its default gateway (SVI) and then be routed
+via the IP-VRF. With EVPN, multiple Layer 2 networks can map to a single IP-VRF.
+
+Let's now look at how a Layer 2 UDN works in practice. For East/West it uses Geneve to ensure MAC-VRF like functionality.
+For North/South by default we leak the Layer 2 into the default VRF (CDN) and use masquerading to avoid collisions with
+other overlapping Layer 2 UDNs. If BGP is enabled, this Layer 2 UDN may be advertised directly (without masquerading) in
+the default VRF. Neither of these options provide IP-VRF like isolation, in both cases the network is leaked into the default
+VRF. Another option is VRF-Lite, where a user can connect a dedicated NIC (think VLAN interface) to the VRF provisioned
+in the Linux host for the Layer 2 UDN. With this model, a Layer 3 VPN is created, but it relies on manual configuration
+on the host, as well as configuration on the upstream PE router. Now in order to route Northbound, the pod sends the packet
```

Comment:
AFAIU, this VRF-lite option (L3 VPN) is borderline useless for virtualization workloads since they would require a MAC VRF (which, afaiu, cannot be implemented in vrf-lite).

Does this deserve to be called out in the design doc ?

Reply:
added a sentence